src/csm_testing/tests/sat_functional/hwinv/test_hwinv.py

```diff
@@ -0,0 +1,632 @@
+
+    def test_hwinv_list_cabinet_pdu_power_connectors(self) -> None:
+        """Test that `sat hwinv --list-cabinet-pdu-power-connectors` returns proper header"""
+        command = 'sat hwinv --list-cabinet-pdu-power-connectors'
+        proc = subprocess.run(shlex.split(command), stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=True)
+        stdout = proc.stdout.decode()
+        stderr = proc.stderr.decode()
+
+        self.assertIn("Listing of all cabinet pdu power connectors", stdout)
+        actual_header = get_column_names_list(stdout)
+        expected_header = adjust_expected_header(HWINV_CABINET_PDU_POWER_CONNECTORS_HEADER, stderr)
+
+        if not actual_header:
+            return
+
+        self.assertEqual(expected_header, actual_header)
+
+    def test_hwinv_list_all(self) -> None:
+        """Test that `sat hwinv --list-all` prints all expected sections and their headers."""
+        command = 'sat hwinv --list-all'
+        proc = subprocess.run(shlex.split(command), stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=True)
+        stdout = proc.stdout.decode()
+        stderr = proc.stderr.decode()
+
+        section_headers = {
+            "Listing of all nodes": HWINV_NODE_HEADER,
+            "Listing of all chassis": HWINV_CHASSIS_HEADER,
+            "Listing of all HSN boards": HWINV_HSN_BOARDS_HEADER,
+            "Listing of all compute modules": HWINV_COMPUTE_MODULES_HEADER,
+            "Listing of all router modules": HWINV_ROUTER_MODULES_HEADER,
+            "Listing of all node enclosures": HWINV_NODE_ENCLOSURES_HEADER,
+            "Listing of all node enclosure power supplies": HWINV_NODE__ENCLOSURE_POWER_SUPPLIES_HEADER,
+            "Listing of all processors": HWINV_PROCS_HEADER,
+            "Listing of all node accelerators": HWINV_NODE_ACCELS_HEADER,
+            "Listing of all node accelerator risers": HWINV_NODE_ACCEL_RISERS_HEADER,
+            "Listing of all node HSN NICS": HWINV_NODE_HSN_NICS_HEADER,
+            "Listing of all memory modules": HWINV_MEMS_HEADER,
+            "Listing of all drives": HWINV_DRIVES_HEADER,
+            "Listing of all CMM rectifiers": HWINV_CMM_RECTIFIERS_HEADER,
+            "Listing of all node bmcs": HWINV_NODE_BMCS_HEADER,
+            "Listing of all router bmcs": HWINV_ROUTER_BMCS_HEADER,
+            "Listing of all mgmt switches": HWINV_MGMT_SWITCHES_HEADER,
+            "Listing of all cabinet pdus": HWINV_CABINET_PDUS_HEADER,
+            "Listing of all cabinet pdu power connectors": HWINV_CABINET_PDU_POWER_CONNECTORS_HEADER,
+        }
+
+        # Split the output into lines
+        lines = stdout.splitlines()
+        line_count = len(lines)
+
+        # Iterate through lines and process each section
+        for i in range(line_count):
+            line = lines[i].strip()
+
+            if line.startswith("Listing of all ") and line.endswith("################################################################################"):
```

Comment:
This condition is never true because "Listing of all" is on one line, and the "######" string is on the next line, not the same line.